packages/core/src/styles/components/_legend.scss

```diff
@@ -0,0 +1,96 @@
+@import "@carbon/themes/scss/themes";
+
+g.cc-legend {
+	user-select: none;
+
+	g.legend-item {
+		rect.checkbox {
+			stroke-width: 1px;
+			stroke-opacity: 1;
+
+			&:not(.active) ~ g.check {
+				display: none;
+			}
+		}
+
+		g.check > svg path {
+			fill: $ui-background;
+			stroke: $ui-background;
+		}
+	}
+
+	&.clickable {
+		g.legend-item:hover {
+			cursor: pointer;
+		}
+	}
+}
+
+/////////////////////////////////////////////
+///          Theme styles                ////
+/////////////////////////////////////////////
+.carbon--theme--white {
+	g.cc-legend {
+		g.legend-item {
+			rect.checkbox {
+				stroke: $ui-background;
+
+				&:not(.active) {
+					stroke: $text-02;
+					fill: $ui-background;
+				}
+			}
+		}
+	}
+}
+
+.carbon--theme--g10 {
+	@include carbon--theme($carbon--theme--g10) {
+		g.cc-legend {
+			g.legend-item {
+				rect.checkbox {
+					stroke: $ui-background;
+
+					&:not(.active) {
+						stroke: $text-02;
+						fill: $ui-background;
+					}
+				}
+			}
+		}
+	}
+}
+
+.carbon--theme--g90 {
+	@include carbon--theme($carbon--theme--g90) {
+		g.cc-legend {
```

Comment:
Unless there's styles needed at each nesting level, we should collapse these selectors as much as possible

Reply:
mostly a general note

Reply:
ill go through all the styles and collapse empty selectors 👍